src/js/stores/AppsStore.js

```diff
@@ -147,29 +148,47 @@ function processApp(app) {
   return app;
 }
 
-function processEmptyGroup(group) {
-  return Object.assign(processApp({id: group.id}), {
-    isGroup: true,
-    status: null
+function addData(parent, child) {
+  Object.keys(parent).forEach(key => {
+    if (key === "health") {
+      parent[key].forEach((value, index) => {
+        value.quantity += child[key][index].quantity;
+      });
+      return;
+    }
+    parent[key] += child[key];
   });
 }
 
 // TODO: refactor to adhere to /groups endpoint structures.
 // https://github.com/mesosphere/marathon/issues/3383
-function processAppsAndEmptyGroups(group, apps = []) {
-  if (group.apps != null && group.apps.length > 0) {
-    group.apps.forEach(app => {
-      apps.push(processApp(app));
-    });
-  } else if (group.id !== "/") {
-    apps.push(processEmptyGroup(group));
+function processAppsAndGroups(group, root = true) {
```

Comment:
Could you please explain, why it is necessary to have different return values depending on the `root` argument? I can't think of any good reason. 

Let's please adjust the method to always return a list of all items (apps and groups) and the have different store getter that only return a filtered set. Doing so will simplify the implementation and is way more predictable. 
BTW: I'm also ok, with always returning three arrays (items, groups, apps) as long as the method interface is always the same!